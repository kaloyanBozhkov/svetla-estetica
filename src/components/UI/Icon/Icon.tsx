import type { ReactElement } from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

// import icons
import { faSpinner } from '@fortawesome/free-solid-svg-icons'

import styles from './styles.module.scss'

const Icon = ({ icon }: { icon: string }): ReactElement => {
    switch (icon) {
        case 'spinner':
            return <FontAwesomeIcon icon={faSpinner} className={styles.spinner} />

        default:
            return <i>Icon not found</i>
    }
}

export default Icon
