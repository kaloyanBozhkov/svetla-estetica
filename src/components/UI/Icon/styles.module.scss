@import "scss/mixins.scss";

.spinner {
    @include spinAnimation;
}
