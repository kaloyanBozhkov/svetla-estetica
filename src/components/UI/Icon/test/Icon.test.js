import React from 'react'
import { shallow } from 'enzyme'

import Icon from '../Icon'

describe('testing Icon atom', () => {
    it('should render Icon with no icon found', () => {
        expect(shallow(<Icon />)).toMatchSnapshot()
    })
})
