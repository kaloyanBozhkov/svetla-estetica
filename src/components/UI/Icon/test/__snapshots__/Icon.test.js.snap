// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Icon atom should render Icon with no icon found 1`] = `
<i>
  Icon not found
</i>
`;
